### ISSUE TITLE
Minecraft crashing by toggling shaders off and on in chunk with powered create components

### ISSUE THREAD

**Issue by wevellll** [leon-o/iris-flw-compat#176]
Trying to resolve this issue for a while now, everything i've noticed for this moment:

Issue don't have a specific version where it happens, i've tried downgrading and upgrading versions of the compat with iris and sodium, but that's doesn't changing anything. 

Crash happens only if there are powered create components in the chunk(?)

Shaders used: BSL-original version 8.3

Crash log: https://mclo.gs/xL4mabE

Shaders settings: 
[BSLOriginal_v8.3.zip.txt](https://github.com/user-attachments/files/17618514/BSLOriginal_v8.3.zip.txt)

**Comment by wevellll** [comment]
Yeah, there is no actual crash-report, i don't have an idea why, because my game actually crashes, it's a hs_err_pid.log

the latest one: https://mclo.gs/Q1B4cIU

**Comment by Silveere** [comment]
I'm pretty certain this is related. My game doesn't outright crash when I do this, but it seems to cause severe visual bugs. However, I am playing on Linux. I had my partner run the modpack on Windows and try the exact same thing, and his game crashed with the following log: https://mclo.gs/s9FfoPP



<details><summary>Screenshot</summary>

![image](https://github.com/user-attachments/assets/d734b91d-33f6-4681-959b-0bf13628be87)

</details>

<details><summary>Mod list</summary>

<pre><code>Mods:
  [✔] [fabric]ctov-3.4.10
  [✔] ad_astra-fabric-1.20.1-1.15.19
  [✔] AmbientSounds_FABRIC_v6.1.3_mc1.20.1
  [✔] amecs-1.3.10+mc.1.20.1
  [✔] amendments-1.20-1.2.12-fabric
  [✔] appleskin-fabric-mc1.20.1-2.5.1
  [✔] appliedenergistics2-fabric-15.2.12
  [✔] architectury-9.2.14-fabric
  [✔] authme-7.0.2+1.20
  [✔] autotag-2.0.2+1.20
  [✔] badpackets-fabric-0.4.3
  [✔] balm-fabric-1.20.1-7.3.9
  [✔] bclib-3.0.14
  [✔] better-end-4.0.11
  [✔] better-nether-9.0.10
  [✔] BetterF3-7.0.2-Fabric-1.20.1
  [✔] bettermounthud-1.2.2
  [✔] blur-3.1.0
  [✔] bobby-5.0.1
  [✔] borderless-mining-1.1.8+1.20.1
  [✔] botarium-fabric-1.20.1-2.3.4
  [✔] capes-1.5.2+1.20-fabric
  [✔] cc-tweaked-1.20.1-fabric-1.114.0
  [✔] cccbridge-mc1.20.1-fabric-v1.6.2b
  [✔] chat_heads-0.13.6-fabric-1.20
  [✔] Chunky-1.3.146
  [✔] citresewn-1.2.2+1.20.1
  [✔] clientcommands-2.8.5
  [✔] cloth-config-11.1.136-fabric
  [✔] collective-1.20.1-7.87
  [✔] continuity-3.0.0-beta.5+1.20.1
  [✔] create_connected-0.7.4-mc1.20.1+FABRIC+HOTFIX
  [✔] create_enchantment_industry-1.2.16
  [✔] create_power_loader-1.5.3-mc1.20.1-fabric
  [✔] create-fabric-0.5.1-f-build.1417+mc1.20.1
  [✔] createaddition-fabric+1.20.1-1.2.4
  [✔] CreativeCore_FABRIC_v2.12.22_mc1.20.1
  [✔] creeperoverhaul-3.0.2-fabric
  [✔] Cull Less Leaves-1.4.2+1.21-fabric
  [✔] DistantHorizons-2.3.0-b-dev-1.20.1-fabric-forge
  [✔] doubledoors-1.20.1-5.9
  [✔] dynamic-fps-3.7.6+minecraft-1.20.0-quilt
  [✔] dynamiccrosshair-7.4.4+1.20-fabric
  [✔] dynamiccrosshair-compat-3.7.5+1.20.1
  [✔] e4mc_minecraft-fabric-5.2.1
  [✔] effective-2.3.2-1.20.1
  [✔] emi_loot-0.7.4+1.20.1+fabric
  [✔] emi-1.1.18+1.20.1+fabric
  [✔] emiffect-fabric-1.1.2+mc1.20.1
  [✔] emitrades-fabric-1.2.1+mc1.20.1
  [✔] entity_model_features_fabric_1.20.1-2.2.6
  [✔] entity_texture_features_fabric_1.20.1-6.2.8
  [✔] entityculling-fabric-1.7.1-mc1.20.1
  [✔] everycomp-1.20-2.6.91-fabric
  [✔] fabric-carpet-1.20-1.4.112+v230608
  [✔] fabricskyboxes-0.7.3+mc1.20.1
  [✔] fallingleaves-1.15.6+1.20.1
  [✔] FarmersDelight-1.20.1-2.2.0+refabricated
  [✔] fastquit-3.0.0+1.20+
  [✔] ferritecore-6.0.1-fabric
  [✔] figura-0.1.4+1.20.1-fabric-mc
  [✔] friendsandfoes-fabric-mc1.20.1-3.0.6
  [✔] fsb-interop-1.3.6+mc1.20.1-build.50
  [✔] fzzy_config-0.5.7+1.20.1
  [✔] garnished-2.0.2+1.20.1-fabric
  [✔] geckolib-fabric-1.20.1-4.4.9
  [✔] handcrafted-fabric-1.20.1-3.0.6
  [✔] identity-2.7.1-1.20.1-fabric
  [✔] imst-2.1.0
  [✔] Incendium_1.20.x_v5.3.5
  [✔] indium-1.0.34+mc1.20.1
  [✔] InventoryProfilesNext-fabric-1.20-1.10.11
  [✔] iris-1.7.5+mc1.20.1
  [✔] iris-flywheel-compat-fabric1.20.1+1.1.4
  [✔] itemswapper-fabric-0.7.3-mc1.20.1
  [✔] jamlib-0.6.1+1.20.x
  [✔] jei-1.20.1-fabric-15.20.0.105
  [✔] JustEnoughFarmersRecipes-1.20.1-1.0.1
  [✔] krypton-0.2.3
  [✔] kubejs-fabric-2001.6.5-build.16
  [✔] lambdynamiclights-2.3.2+1.20.1
  [✔] language-reload-1.7.1+1.20.1
  [✔] libIPN-fabric-1.20-4.0.2
  [✔] lilac-1.0.0-beta.3
  [✔] litematica-fabric-1.20.1-0.15.4
  [✔] lithium-fabric-mc1.20.1-0.11.2
  [✔] lithostitched-fabric-1.20.1-1.3.4
  [✔] malilib-fabric-1.20.1-0.16.3
  [✔] midnightcontrols-1.9.4+1.20
  [✔] minihud-fabric-1.20.1-0.27.1
  [✔] mixintrace-1.1.1+1.17
  [✔] modmenu-7.2.2
  [✔] moonlight-1.20-2.13.28-fabric
  [✔] more-geodes-1.9.3
  [✔] morechathistory-1.3.0
  [✔] MouseTweaks-fabric-mc1.20-2.26
  [✔] nmuk-1.1.1+mc1.20.1
  [✔] NoChatReports-FABRIC-1.20.1-v2.2.2
  [✔] Nullscape_1.20.x_v1.2.8
  [✔] optigui-2.3.0-beta.6+1.20
  [✔] Ping-Wheel-1.10.0-fabric-1.20.1
  [✔] PresenceFootsteps-1.10.1+1.20.1
  [✔] pride-moths-1.8.5+1.20.1
  [✔] puzzle-1.5.2+1.20
  [✔] qfapi-7.6.0_qsl-6.2.0_fapi-0.92.2_mc-1.20.1
  [✔] quilt_loading_screen-6.1.0+1.20.1
  [✔] quilt-kotlin-libraries-core-5.0.0+kt.2.0.21+flk.1.12.3
  [✔] reeses_sodium_options-1.7.2+mc1.20.1-build.101
  [✔] replaymod-1.20.1-2.6.20
  [✔] resourcefulconfig-fabric-1.20.1-2.1.2
  [✔] resourcefullib-fabric-1.20.1-2.1.29
  [✔] rhino-fabric-2001.2.3-build.6
  [✔] right-click-harvest-3.2.3+1.19.x-1.20.1-fabric
  [✔] screenshot-to-clipboard-1.0.9-fabric
  [✔] sliceanddice-fabric-3.3.1
  [✔] sodium-extra-0.5.4+mc1.20.1-build.115
  [✔] sodium-fabric-0.5.11+mc1.20.1
  [✔] sound-physics-remastered-fabric-1.20.1-1.4.5
  [✔] spark-1.10.53-fabric
  [✔] Steam_Rails-1.6.4+fabric-mc1.20.1
  [✔] supplementaries-1.20-3.1.8-fabric
  [✔] Terralith_1.20.x_v2.5.4
  [✔] treeharvester-1.20.1-9.1
  [✔] tweakeroo-fabric-1.20.1-0.17.1
  [✔] way2wayfabric-0.0.6
  [✔] waystones-fabric-1.20-14.1.6
  [✔] whereisit-2.6.3+1.20.1
  [✔] wthit-quilt-8.15.3
  [✔] Xaeros_Minimap_24.6.1_Fabric_1.20
  [✔] XaerosWorldMap_1.39.0_Fabric_1.20
  [✔] YetAnotherConfigLib-3.6.1+1.20.1-fabric
  [✔] yosbr-0.1.2
  [✔] YungsApi-1.20-Fabric-4.0.6
  [✔] YungsBetterDesertTemples-1.20-Fabric-3.0.3
  [✔] YungsBetterDungeons-1.20-Fabric-4.0.4
  [✔] YungsBetterJungleTemples-1.20-Fabric-2.0.5
  [✔] YungsBetterMineshafts-1.20-Fabric-4.0.4
  [✔] YungsBetterNetherFortresses-1.20-Fabric-2.0.6
  [✔] YungsBetterOceanMonuments-1.20-Fabric-3.0.4
  [✔] YungsBetterStrongholds-1.20-Fabric-4.0.3
  [✔] YungsBetterWitchHuts-1.20-Fabric-3.0.3
  [✔] YungsBridges-1.20-Fabric-4.0.3
  [✔] YungsExtras-1.20-Fabric-4.0.3
  [✔] Zoomify-2.14.2+1.20.1
</code></pre>
</details>

**Comment by darceno** [comment]
I'm having the same problem, in both Fabric & Forge. The last lines of latest.log are:

```
[24nov.2024 12:46:53.531] [Render thread/INFO] [Oculus/]: Destroying pipeline NamespacedId{namespace='minecraft', name='overworld'}
[24nov.2024 12:46:53.576] [Render thread/WARN] [Oculus/]: Found flag CUSTOM_IMAGES
[24nov.2024 12:46:53.576] [Render thread/WARN] [Oculus/]: Found flag BLOCK_EMISSION_ATTRIBUTE
[24nov.2024 12:46:53.576] [Render thread/WARN] [Oculus/]: Tried to get boolean value for unknown option: 1, defaulting to true!
[24nov.2024 12:46:53.576] [Render thread/WARN] [Oculus/]: Tried to get boolean value for unknown option: 1, defaulting to true!
[24nov.2024 12:46:53.576] [Render thread/WARN] [Oculus/]: Tried to get boolean value for unknown option: 1, defaulting to true!
[24nov.2024 12:46:53.576] [Render thread/WARN] [Oculus/]: Tried to get boolean value for unknown option: 1, defaulting to true!
[24nov.2024 12:46:53.576] [Render thread/WARN] [Oculus/]: Tried to get boolean value for unknown option: 1, defaulting to true!
[24nov.2024 12:46:53.576] [Render thread/WARN] [Oculus/]: Tried to get boolean value for unknown option: 1, defaulting to true!
[24nov.2024 12:46:53.578] [Render thread/INFO] [Oculus/]: Profile: MEDIUM (+2 options changed by user)
[24nov.2024 12:46:54.169] [Render thread/INFO] [Oculus/]: Using shaderpack: ComplementaryReimagined_r5.3 + EuphoriaPatches_1.4.3
[24nov.2024 12:46:54.169] [Render thread/INFO] [Oculus/]: Creating pipeline for dimension NamespacedId{namespace='minecraft', name='overworld'}
[24nov.2024 12:46:54.171] [Render thread/INFO] [Oculus/]: Starting custom uniform resolving
[24nov.2024 12:46:55.185] [Render thread/INFO] [ChunkBuilder/]: Stopping worker threads
[24nov.2024 12:46:55.191] [Render thread/INFO] [ChunkBuilder/]: Started 6 worker threads
[24nov.2024 12:46:55.201] [Render thread/INFO] [net.minecraft.client.gui.components.ChatComponent/]: [System] [CHAT] Sombreadores alternados para ComplementaryReimagined_r5.3 + EuphoriaPatches_1.4.3!
[24nov.2024 12:46:55.370] [Render thread/WARN] [irisflw/]: GLSL version is lower than 330, set to 330
[24nov.2024 12:46:55.392] [Render thread/WARN] [irisflw/]: GLSL version is lower than 330, set to 330
```

**Comment by wevellll** [comment]
Are you using Complementary shaders and it still happens? If so, it's even bigger issue, for all shaderpacks i suppose, or the majority.

**Comment by darceno** [comment]
@wevellll yes, I was using Complementary and also tried with other shaders, all with the same result.

Fortunately, downgrading iris-flw-compat to 1.1.2 fixes this issue for me.

**Comment by wevellll** [comment]
i've tried downgrading a while ago i don't remember the version i've downgraded for, but it didn't solved anything. so my solution was just don't play with shaders, at least by now, when the issue is still wasn't fixed.

**Comment by leon-o** [comment]
Duplicate of #236. Closing in favor of the original issue.
